frontend/src/views/ManagementPage/PublicationPreviewDialog.vue

```diff
@@ -0,0 +1,50 @@
+<template>
+  <v-dialog persistent max-width="1000" v-model="value" @click:outside="$emit('close')">
+    <v-card>
+      <div class="mb-4 pa-4 d-flex" style="background-color: #F5F5F5">
+        <v-spacer></v-spacer>
+        <v-btn color="primary" outlined @click="$emit('close')">
+          Fermer
+        </v-btn>
+      </div>
+      <v-card-text>
+        <CanteenDashboard :diagnostics="diagnostics" />
+      </v-card-text>
+    </v-card>
+  </v-dialog>
+</template>
+
+<script>
+import CanteenDashboard from "@/components/CanteenDashboard"
+import Constants from "@/constants"
+
+export default {
+  name: "PublicationPreview",
+  components: { CanteenDashboard },
+  props: {
+    canteen: {
+      type: Object,
+      required: true,
+    },
+    value: {
+      type: Boolean,
+      required: true,
+    },
+  },
+  computed: {
+    diagnostics() {
+      const diagnostics = this.canteen.diagnostics
+      return {
+        previous:
+          diagnostics.find((x) => x.year === 2019) || Object.assign({}, Constants.DefaultDiagnostics, { year: 2019 }),
+        latest:
+          diagnostics.find((x) => x.year === 2020) || Object.assign({}, Constants.DefaultDiagnostics, { year: 2020 }),
```

Comment:
Si je remplis les valeurs (par exemple) pour le diagnostic, mais pas des `communicationSupports` je reçois `TypeError: _vm.latestDiagnostic.communicationSupports is null` quand j'essaye de ouvrir ce dialog. Je pense qu'on n'a pas le même defauts dans le BDD et dans le constants. C'est pas forcement un problème avec le code celui-ci mais je note quand même

Reply:
Ah mince, c'est possible que ça vienne du fait que _communicationsSupports_ est un tableau non ? peut-être devrait-on envoyer un tableau vide ? `[]`

Reply:
Hmm bizarre... on a bien le tableau vide comme valeur par défaut dans constants.js :thinking:

Reply:
Je prends :point_up:

Reply:
Fixé ici : https://github.com/betagouv/ma-cantine/pull/308/commits/8813239a15e296790bdb3ff261bf42432ce56c52